src/audio/module_adapter/module_adapter.c

```diff
@@ -105,10 +105,10 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	if (drv->type == SOF_COMP_MODULE_ADAPTER) {
 		struct ipc_config_process *ipc_module_adapter = spec;
 
-		dst->data = ipc_module_adapter->data;
+		dst->init_data = ipc_module_adapter->data;
 		dst->size = ipc_module_adapter->size;
 	} else {
-		dst->data = spec;
+		dst->init_data = spec;
```

Comment:
@lyakh i think this needs to go a bit further. WIth the module adapter there are 2 ways the inital data gets used. Modules like the volume use this in their init callback to read the IPC payload. But modules like the cadence codec, use the dst->data to apply the config to just prior to prepare(). So if you change to initi_data, you'll need to change the codec implementation too.

Reply:
ah, there's a problem there: in that `else` clause above the size isn't known.

Reply:
I'm not certain about the case under `else` clause, `drv->type` is not `SOF_COMP_MODULE_ADAPTER` during `module_adapter_new()`? Is it specific for `SOF_COMP_VOLUME`? @ranj063 

Anyway IMO this might cause potential issues while `dst->data` is assigned without `dst->size`

Reply:
@ranj063 do you mean https://github.com/thesofproject/sof/blob/adce54445012726ee559a307355ac31e46b25722/src/audio/module_adapter/module/cadence.c#L243 ? So that line should become
```
ret = memcpy_s(setup_cfg->data, setup_cfg->size, codec->cfg.init_data, setup_cfg->size);
```
Any more such places? Cadence isn't tested by cmocka or testbench, is it?